Support init script at JdbcDatabaseContainer level

AnkBurov opened the pull request:
Currently init script can be applied only through ContainerDatabaseDriver which doesn't support private Docker registries. This limitation also denies generic container approach of usage. 
More information about the problem is in the corresponding issue: https://github.com/testcontainers/testcontainers-java/issues/532

So I've implemented an init script support for all Jdbc containers. 

Resolve #532

AnkBurov commented on modules/database-commons/src/main/java/org/testcontainers/ext/ScriptUtils.java:
```diff
@@ -210,6 +214,30 @@ public static boolean containsSqlScriptDelimiters(String script, String delim) {
+	 * @param databaseDelegate database delegate for script execution
+	 * @param initScriptPath   the resource to load the init script from
+	 */
+	public static void runInitScript(DatabaseDelegate databaseDelegate, String initScriptPath) {
```
This method can also be used in ContainerDatabaseDriver.runInitScriptIfRequired(...) to replace almost all of the method's code, but I didn't do it to have diff smaller.

kiview commented:
Not really your fault, but I just realised the intendation is this file is totally messed up.

kiview commented on modules/database-commons/src/main/java/org/testcontainers/ext/ScriptUtils.java:
```diff
@@ -210,6 +214,30 @@ public static boolean containsSqlScriptDelimiters(String script, String delim) {
+	 */
+	public static void runInitScript(DatabaseDelegate databaseDelegate, String initScriptPath) {
+		try {
+			URL resource = Thread.currentThread().getContextClassLoader().getResource(initScriptPath);
```
Why do you need to use `Thread` here and can't use `ScriptUtils.class.getResource("foo")`?

AnkBurov replied:
It's an old code that previously was in ContainerDatabaseDriver.runInitScriptIfRequired(...). It's because module 'jdbc-test' is a separate module and it has it's own classpath.

Hmm, I wonder will init scripting work with testng and separate runner. Need to check.

AnkBurov replied:
I think using classloader resources will be enough here.

AnkBurov commented:
Added new test to SimplePostgreSQLTest and refactored another - to have tests completely isolated and not have several postgresql containers (if in second test would use postresql.withInitScript as local variable).

AnkBurov commented on modules/jdbc-test/src/test/java/org/testcontainers/junit/SimplePostgreSQLTest.java:
```diff
@@ -17,23 +17,42 @@
+        }
+    }
+
+    private ResultSet performQuery(JdbcDatabaseContainer container, String sql) throws SQLException {
```
I have a strong desire to extract this code to separate TestUtils and use it here and in SimpleMySQLTest.

Also why no connections and result sets here and in SimpleMySQLTest are closes? Of course it's all tests, but still.

rnorth commented:
We're shortly going to be merging #574, which changes our build system to Gradle. This is in part intended to make contributions of modules easier (per #564), but unfortunately means that for a short while your PR is going to show merge conflicts with the master branch.

I just want to let you know we don't want to create new work for you: *we'll take care of the merge conflicts shortly*. Please don't worry - we're grateful for your PR and want to help integrate it soon. Thank you.

rnorth commented:
Sorry this has waited so long - I'm not sure why we haven't just merged it yet. I'll assign to myself and will aim to fix the merge conflict and push this through soon.
